doc/source/serve/multi-app.md

```diff
@@ -0,0 +1,208 @@
+# Deploying Multiple Serve Applications
+
+In Ray 2.4+, deploying multiple independent Serve applications is supported. This user guide walks through how to generate a multi-application config file and deploy it using the Serve CLI, and monitor your applications using the CLI and the Ray Serve dashboard.
+
+## When to Use Multiple Applications
+It can be useful to deploy multiple independent applications on one Ray cluster. However, it is also important to keep in mind the downsides of doing so.
+
+We suggest you deploy multiple applications per cluster if:
+* You want to save cost by sharing resources between applications.
+* You have many small independent models, and you want to be able to easily add and delete models often.
+
+However, we suggest you limit to one application per cluster if:
+* Your applications are large and require a lot of resources. Deploying multiple applications per cluster can cause competition of resources in this case.
+* You want better isolation between applications. Although applications are independent of one another, isolation can't be guaranteed. Also, if you need to change the configuration of the cluster for one application, all other applications will also be affected. 
+
+
+## Get Started
+
+Define a Serve application:
+```{literalinclude} doc_code/basic_calculator.py
+:language: python
+:start-after: __serve_example_begin__
+:end-before: __serve_example_end__
+```
+
+Copy this to a file named `calculator.py`.
+
+Define a second Serve application:
+```{literalinclude} doc_code/basic_greet.py
+:language: python
+:start-after: __serve_example_begin__
+:end-before: __serve_example_end__
+```
+Copy this to a file named `greet.py`.
+
+Generate a multi-application config file that contains both of these two applications and save it to `config.yaml`.
+
+```
+serve build --multi-app calculator:app greet:app -o config.yaml
+```
+
+This generates the following config:
+```yaml
+proxy_location: EveryNode
+http_options:
+  host: 0.0.0.0
+  port: 8000
+
+applications:
+
+- name: app1
+  route_prefix: /calculator
+  import_path: calculator:app
+  runtime_env: {}
+  deployments:
+  - name: Multiplier
+  - name: Adder
+  - name: Router
+  - name: DAGDriver
+
+- name: app2
+  route_prefix: /greet
+  import_path: greet:app
+  runtime_env: {}
+  deployments:
+  - name: greet
+  - name: DAGDriver
+```
+
+:::{note} 
+The names for each application are auto-generated as `app1`, `app2`, etc. To give custom names to the applications, modify the config file before moving on to the next step.
+:::
+
+### Deploy the Applications
+To deploy the applications, be sure to start a Ray cluster first.
```

Comment:
Should we add the serve run example here too?

Reply:
Added a section on easy development with `serve run`